### PR TITLE
vector streams

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
- [ ] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:

<TEXT>

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Vektor Chunks Stream nach Python](https://trello.com/c/OvgxPSsg/1194-vektor-chunks-stream-nach-python)

File: datatypes/src/collections/ipc.rs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+impl<'g, G> FeatureCollectionIpc<'g> for FeatureCollection<G>
+where
+    Self: IntoGeometryOptionsIterator<'g>,
+{
+    fn to_arrow_record_batch(&'g self) -> Result<arrow::record_batch::RecordBatch> {
+        // Currently, PyArrow does not support our storage of geometries
+        // Thus, we will transform them to WKTs.
+        // TODO: use `let record_batch: RecordBatch = struct_ref.into();` instead
```
Comment (jdroenner) [inline datatypes/src/collections/ipc.rs:28]: can you add some information when to switch that?

File: datatypes/src/collections/ipc.rs [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            if let Some(geometry) = geometry_option {
+                wkts.push(geometry.wkt_string());
+            } else {
+                wkts.push(String::new());
+            }
+        }
+        let wkt_array = Arc::new(arrow::array::StringArray::from(wkts));
+
+        let arrow::datatypes::DataType::Struct(fields) = self.table.data_type() else {
+            unreachable!("unable to get datatype as struct")
```
Comment (jdroenner) [inline datatypes/src/collections/ipc.rs:41]: shouldn't this tell why it is unreachable?

File: services/src/workflows/vector_stream.rs [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    fn finished(&mut self, ctx: &mut Self::Context) {
+        ctx.stop();
+
+        self.abort_processing();
+    }
+
+    fn handle(&mut self, msg: Result<ws::Message, ws::ProtocolError>, ctx: &mut Self::Context) {
+        match msg {
+            Ok(ws::Message::Ping(msg)) => ctx.pong(&msg),
+            Ok(ws::Message::Text(text)) if &text == "NEXT" => self.next_tile(ctx),
```
Comment (jdroenner) [inline services/src/workflows/vector_stream.rs:50]: can we name it next_chunk or something like this?

**Review by jdroenner — CHANGES_REQUESTED** [review]

File: datatypes/src/collections/ipc.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply datatypes/src/collections/ipc.rs:28]: I don't know, I didn't find any information on that. It just did not work.

File: services/src/workflows/vector_stream.rs (hunk 3 above)
Comment (ChristianBeilschmidt) [reply services/src/workflows/vector_stream.rs]: [45ac504](https://github.com/geo-engine/geoengine/pull/747/commits/45ac50450977748902ee17981cdfd5d7b0327592)

**Review by jdroenner — APPROVED** [review]

**Comment by jdroenner** [comment]
bors merge

**Pull request merged**